File: PathoFroh/src/main/java/com/patho/main/repository/service/MaterialPresetRepositoryCustom.java
package com.patho.main.repository.service;

import com.patho.main.model.MaterialPreset;

import java.util.List;

public interface MaterialPresetRepositoryCustom {
	public List<MaterialPreset> findAll(boolean loadStainings);

	public List<MaterialPreset> findAllByName(String name, boolean loadStainings);

	public List<MaterialPreset> findAllIgnoreArchivedOrderByPriorityCountDesc(boolean loadStainings,
			boolean irgnoreArchived);
}

File: PathoFroh/src/main/kotlin/com/patho/main/repository/jpa/custom/MaterialPresetRepositoryCustom.kt
package com.patho.main.repository.jpa.custom

import com.patho.main.model.preset.MaterialPreset
import com.patho.main.model.preset.StainingPrototype


interface MaterialPresetRepositoryCustom {

    fun findAll(loadStainings: Boolean): List<MaterialPreset>

    fun findAllByName(name: String, loadStainings: Boolean): List<MaterialPreset>

    fun findAllOrderByPriorityCountDesc(loadStainings: Boolean,
                                        irgnoreArchived: Boolean): List<MaterialPreset>


    fun findAllOrderByIndexInListAsc(loadStainings: Boolean,
                                       irgnoreArchived: Boolean) : List<MaterialPreset>
}